src/plugins/dashboard/public/application/top_nav/show_share_modal.tsx

```diff
@@ -104,21 +110,50 @@ export function ShowShareModal({
     );
   };
 
+  const rawDashboardState = stateToRawDashboardState({
+    state: currentDashboardState,
+    version: kibanaVersion,
+  });
+
+  const locatorParams: DashboardAppLocatorParams = {
+    dashboardId: savedDashboard.id,
+    filters: rawDashboardState.filters,
+    preserveSavedFilters: true,
+    query: rawDashboardState.query,
+    savedQuery: rawDashboardState.savedQuery,
+    useHash: false,
+    panels: rawDashboardState.panels,
+    timeRange,
+    viewMode: ViewMode.VIEW, // For share locators we always load the dashboard in view mode
+    refreshInterval: undefined, // We don't share refresh interval externally
+    options: rawDashboardState.options,
+  };
+
   share.toggleShareContextMenu({
     isDirty,
     anchorElement,
     allowEmbed: true,
     allowShortUrl: dashboardCapabilities.createShortUrl,
     shareableUrl: setStateToKbnUrl(
       '_a',
-      stateToRawDashboardState({ state: currentDashboardState, version: kibanaVersion }),
+      rawDashboardState,
       { useHash: false, storeInHashQuery: true },
       unhashUrl(window.location.href)
     ),
     objectId: savedDashboard.id,
     objectType: 'dashboard',
     sharingData: {
-      title: savedDashboard.title,
+      title:
+        savedDashboard.title ||
+        i18n.translate('dashboard.share.defaultDashboardTitle', {
+          defaultMessage: 'Dashboard [{date}]',
+          values: { date: moment().toISOString(true) },
```

Comment:
Does this date string need to be  converted to the user's timezone and formatted per the time format in Advanced Settings?

Reply:
That's a great point @tsullivan ! Do you mind if I address this in a follow up PR as I did this same thing in discover

https://github.com/elastic/kibana/blob/f0bf87768622f13e4d1163e37094f50c1e5ba871/src/plugins/discover/public/application/apps/main/components/top_nav/get_top_nav_links.ts#L129